src/Microsoft.AspNet.Security/AuthenticationTicket.cs

```diff
@@ -29,11 +23,34 @@ public AuthenticationTicket(ClaimsIdentity identity, AuthenticationProperties pr
         }
 
         /// <summary>
+        /// Initializes a new instance of the <see cref="AuthenticationTicket"/> class
+        /// </summary>
+        /// <param name="identity">the <see cref="ClaimsPrincipal"/> that represents the authenticated user.</param>
+        /// <param name="properties">additional properties that can be consumed by the user or runtims.</param>
+        /// <param name="authenticationType">the authentication middleware that was responsible for this ticket.</param>
+        public AuthenticationTicket(ClaimsPrincipal principal, AuthenticationProperties properties, string authenticationType)
+        {
+            AuthenticationType = authenticationType;
+            Principal = principal;
+            Properties = properties ?? new AuthenticationProperties();
+        }
+
+        /// <summary>
+        /// Gets the authenticated user identity.
+        /// </summary>
+        public string AuthenticationType { get; private set; }
+
+        /// <summary>
         /// Gets the authenticated user identity.
         /// </summary>
         public ClaimsIdentity Identity { get; private set; }
 
         /// <summary>
+        /// Gets the authenticated user identity.
+        /// </summary>
+        public ClaimsPrincipal Principal{ get; private set; }
```

Comment:
We really don't want two disconnected identity fields here. They either need to be connected or there should only be one.

Nit: spacing.